demos/speculation_over_syscall.cc

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __linux__
+#  error Unsupported OS. Linux required.
+#endif
+
+#ifndef __aarch64__
+#  error Unsupported architecture. ARM64 required.
+#endif
+
+#include <array>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+
+#include <signal.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "cache_sidechannel.h"
+#include "instr.h"
+
+const char *public_data = "Hello, world!";
```

Comment:
This is another demo where the architectural path isn't an access -- it's a fault. So this would be a place to use `AddHitAndComputeScores` and remove `public_data` to reduce the number of moving parts at toplevel.

My mental model is that demos with `public_data` are only "supposed" to access that, but end up accessing `private_data` on the nonarchitectural path.

Reply:
I tested that and it didn't work even in this case. If you are interested you can try to debug it using your Raspberry PI or on Meltdown-US with an Intel computer (booted with pti=off on the kernel command line). I created an issue for that: https://github.com/google/safeside/issues/50

Reply:
Otherwise your mental model is correct. It's probably a problem with the implementation.